content/blog/tips-for-faster-ci-builds/index.md

```diff
@@ -0,0 +1,218 @@
++++
+title = "Tips for Faster Rust CI Builds"
+date = 2025-01-18
+draft = false
+template = "article.html"
+[extra]
+series = "Rust Insights"
+resources = [
+    "[Fast Rust Builds by Matklad](https://matklad.github.io/2021/09/04/fast-rust-builds.html#CI-Workflow)"
+]
+hero = "hero.svg"
++++
+
+I've been working with many clients lately who host their Rust projects on GitHub.
+CI is typically a bottleneck in the development process since it can significantly slow down feedback loops.
+However, there are several effective ways to speed up your GitHub Actions workflows!
+
+{% info(title="Want a Real-World Example?", icon="crab") %}
+
+Check out this production-ready GitHub Actions workflow that implements all the tips from this article:
+[click here](https://github.com/lycheeverse/lychee/blob/master/.github/workflows/ci.yml).
+
+{% end %}
+
+## Use Swatinem's cache action
+
+This is easily my most important recommendation on this list.
+
+My friend [Arpad Borsos](https://github.com/swatinem), also known as Swatinem, has created a cache action specifically tailored for Rust projects.
+It's an excellent way to speed up any Rust CI build and requires no code changes to your project.
+
+```yaml
+name: CI
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest-arm64
+
+    env:
+      CARGO_TERM_COLOR: always
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+
+      # The secret sauce!
+      - uses: Swatinem/rust-cache@v2
+
+      - run: |
+          cargo check
+          cargo test
+          cargo build --release
+```
+
+The action requires no additional configuration and works out of the box.
+There's no need for a separate step to store the cache — this happens automatically through a post-action.
+This approach ensures that broken builds aren't cached, and for subsequent builds, you can save several minutes of build time.
+
+Here's the [documentation](https://github.com/Swatinem/rust-cache) where you can learn more. 
+
+## Use the `--locked` flag
+
+When running `cargo build`, `cargo test`, or `cargo check`, you can pass the `--locked` flag to prevent Cargo from updating the `Cargo.lock` file.
+
+This is particularly useful for CI operations since you save the time to update dependencies.
+Typically you want to test the exact dependency versions specified in your lock file anyway.
```

Comment:
Might be a bit too short? I'm missing a bit of background information why you'd want `--locked` anyway.